src/main/java/com/google/devtools/build/lib/remote/BUILD

```diff
@@ -49,3 +58,17 @@ java_library(
         "@googleapis//:google_watch_v1_java_proto",
     ],
 )
+
+java_library(
+    name = "remote-utils",
```

Comment:
Since I see no other example in Bazel code doing this, I suspect it won't fly -- you'd need to properly separate this into remote.utils subdirectory (which I think is a good idea anyway). Another way to do it is just keep it all together in the remote library, and not separate the logging, which is much simpler.
Also, important note: in order for all this to work internally, you will need to amend the internal BUILD files as well. I strongly suspect that the automatic merge process of external patches will not do it. I would suggest you just send a CL for this internally instead, that would not only take care of this, but also run all the internal tests.

Reply:
Asked around for a good way to do this.
A good tool is this one: devtools/blaze/bazel/admin/import-patch.sh

For example, for your change you can run this:
devtools/blaze/bazel/admin/import-patch.sh https://github.com/bazelbuild/bazel/pull/4669

It will generate a cl by the automatic merge process. You can then go into that cl to see how the changes map, and to run presubmits.

Reply:
I've factored out util classes from the remote package to remote.util. I think this will probably be the last commit in this PR, which I think I will continue internally instead because of the aforementioned reasons. If this sounds reasonable to you, I I'll close this.